### PR TITLE
[ML][Pipelines] Enable AutoML DSL tests

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

Enable AutoML DSL tests skipped during GitHub migration.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py [hunk 1]
```diff
@@ -59,9 +63,10 @@ def train_automl_classification_in_pipeline(
         )
 
         pipeline_job: PipelineJob = train_automl_classification_in_pipeline(class_train, class_valid)
+        pipeline_job.settings.default_compute = "cpu-cluster"
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py:66]: may add a variable for default_compute

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py [hunk 2]
```diff
@@ -105,9 +106,10 @@ def train_automl_regression_in_pipeline(regression_train_data):
         )
 
         pipeline_job: PipelineJob = train_automl_regression_in_pipeline(regression_train)
+        pipeline_job.settings.default_compute = "cpu-cluster"
 
         from_rest_pipeline_job = client.jobs.create_or_update(pipeline_job)
-        assert_final_job_status(from_rest_pipeline_job, client, PipelineJob, JobStatus.COMPLETED)
+        cancel_job(client, from_rest_pipeline_job)
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py:112]: how long does it take to run these automl tests in live mode?

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py (hunk 1 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py]: added for `cpu-cluster` and `gpu-cluster`

File: sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py (hunk 2 above)
Comment (zhengfeiwang) [reply sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_automl_dsl_pipeline.py:116]: I have tested locally on canary region, it takes around 7.5 minutes to run all these tests
image

**Review by brynn-code — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**